### ISSUE TITLE
Crash compiling libunwind.cpp

### ISSUE THREAD

**Issue by brooksdavis** [CTSRD-CHERI/llvm-project#549]
With master at ea3e23aa96b1e39e8473d4e93b773243ba910e58 I get this crash
```
clang++: /home/bed22/cheri/llvm-project/llvm/lib/IR/Value.cpp:696: const llvm::Value* llvm::Value::stripAndAccumulateConstantOffsets(const llvm::DataLayout&, llvm::APInt&, bool, llvm::function_ref<bool(llvm::Value&, llvm::APInt&)>) const: Assertion `BitWidth == DL.getIndexTypeSizeInBits(getType()) && “The offset bit width does not match the DL specification.“' failed.
```
The reduced reproducer is:
```
// RUN: %clang_cc1 -triple riscv64-unknown-freebsd13.0 --mrelax-relocations -fhalf-no-semantic-interposition -target-feature +m -target-feature +a -target-feature +f -target-feature +d -target-feature +c -target-feature +xcheri -target-feature -relax -target-feature +cap-mode -target-feature -save-restore -target-abi l64pc128d -msmall-data-limit 0 -fcoverage-compilation-dir=/local/scratch/bed22/cheri/build/cheribsd-riscv64-purecap-build/local/scratch/bed22/cheri/cheribsd/riscv.riscv64c/lib/libgcc_eh -O2 -fno-signed-char -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o --emit-llvm -disable-O0-optnone -Wimplicit-int  -Wfatal-errors %s
class a {
public:
  a(void *);
};
class b {
public:
  static b c;
};
template <typename d, typename e> class f {
public:
  f(int *, d &);
  virtual d g();
  e h;
};
template <typename d, typename e> f<d, e>::f(int *j, d &) : h(j) { g(); }
int i;
void k() { f<b, a>(&i, b::c); }
```

**Comment by brooksdavis** [comment]
More crash context:
```
===> lib/libgcc_s (obj,all,install) 
clang++: /home/bed22/cheri/llvm-project/llvm/lib/IR/Value.cpp:696: const llvm::Value* llvm::Value::stripAndAccumulateConstantOffsets(const llvm::DataLayout&, llvm::APInt&, bool, llvm::function_ref<bool(llvm::Value&, llvm::APInt&)>) const: Assertion `BitWidth == DL.getIndexTypeSizeInBits(getType()) && "The offset bit width does not match the DL specification."' failed.
PLEASE submit a bug report to https://github.com/CTSRD-CHERI/llvm-project/issues and include the crash backtrace, preprocessed source, and associated run script.
If you built clang from source, please provide a reduced test case by running:
        $LLVM_BUILD_DIR/bin/creduce_crash_testcase.py <reproducer>.sh
Stack dump:
0.      Program arguments: /home/bed22/cheri/output/sdk/bin/clang++ -fuse-ld=lld -Qunused-arguments -target riscv64-unknown-freebsd13.0 --sysroot=/local/scratch/bed22/cheri/build/cheribsd-riscv64-purecap-build/local/scratch/bed22/cheri/cheribsd/riscv.riscv64c/tmp -B/local/scratch/bed22/cheri/build/cheribsd-riscv64-purecap-build/local/scratch/bed22/cheri/cheribsd/riscv.riscv64c/tmp/usr/bin -fvisibility=hidden -fPIC -O2 -pipe -fno-common -march=rv64imafdcxcheri -mabi=l64pc128d -D__LP64__=1 -I/local/scratch/bed22/cheri/cheribsd/contrib/subrepo-cheri-libunwind/include -I/local/scratch/bed22/cheri/cheribsd/lib/libgcc_eh -D_LIBUNWIND_IS_NATIVE_ONLY -D_LIBUNWIND_USE_FRAME_HEADER_CACHE -I/local/scratch/bed22/cheri/cheribsd/lib/msun/src -g -mno-relax -MD -MF.depend.libunwind.o -MTlibunwind.o -Wno-format-zero-length -nobuiltininc -idirafter /local/scratch/bed22/cheri/output/sdk/lib/clang/13.0.0/include -Werror=implicit-function-declaration -Wsystem-headers -Werror -Wall -Wno-format-y2k -Wno-uninitialized -Wno-error=pass-failed -Wno-empty-body -Wno-string-plus-int -Wno-unused-const-variable -Wno-tautological-compare -Wno-unused-value -Wno-parentheses-equality -Wno-unused-function -Wno-enum-conversion -Wno-unused-local-typedef -Wno-address-of-packed-member -Wno-switch -Wno-switch-enum -Wno-knr-promoted-parameter -Qunused-arguments -fno-rtti -std=c++11 -Wno-error=non-c-typedef-for-linkage -Wno-c++11-extensions -fno-exceptions -funwind-tables -c /local/scratch/bed22/cheri/cheribsd/contrib/subrepo-cheri-libunwind/src/libunwind.cpp -o libunwind.o
1.      <eof> parser at end of file
2.      Optimizer
 #0 0x0000564531aed6bc llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/bed22/cheri/output/sdk/bin/clang+++0x27b06bc)
 #1 0x0000564531aeb384 llvm::sys::RunSignalHandlers() (/home/bed22/cheri/output/sdk/bin/clang+++0x27ae384)
 #2 0x0000564531aeb605 llvm::sys::CleanupOnSignal(unsigned long) (/home/bed22/cheri/output/sdk/bin/clang+++0x27ae605)
 #3 0x0000564531a4c8a8 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #4 0x00007fd5ae412980 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12980)
 #5 0x00007fd5ad2edfb7 raise /build/glibc-S9d2JN/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #6 0x00007fd5ad2ef921 abort /build/glibc-S9d2JN/glibc-2.27/stdlib/abort.c:81:0
 #7 0x00007fd5ad2df48a __assert_fail_base /build/glibc-S9d2JN/glibc-2.27/assert/assert.c:89:0
 #8 0x00007fd5ad2df502 (/lib/x86_64-linux-gnu/libc.so.6+0x30502)
 #9 0x0000564531486930 llvm::Value::stripAndAccumulateConstantOffsets(llvm::DataLayout const&, llvm::APInt&, bool, llvm::function_ref<bool (llvm::Value&, llvm::APInt&)>) const (/home/bed22/cheri/output/sdk/bin/clang+++0x2149930)
#10 0x0000564531b2518c llvm::tryPromoteCall(llvm::CallBase&) (/home/bed22/cheri/output/sdk/bin/clang+++0x27e818c)
#11 0x0000564531572118 llvm::InlinerPass::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x2235118)
#12 0x0000564531573831 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::InlinerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x2236831)
#13 0x0000564533bfe06d llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x48c106d)
#14 0x0000564531569ab1 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x222cab1)
#15 0x0000564533c00cd8 llvm::DevirtSCCRepeatedPass::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x48c3cd8)
#16 0x0000564531569a71 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::DevirtSCCRepeatedPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/bed22/cheri/output/sdk/bin/clang+++0x222ca71)
#17 0x0000564533bff1df llvm::ModuleToPostOrderCGSCCPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x48c21df)
#18 0x0000564531569af1 llvm::detail::PassModel<llvm::Module, llvm::ModuleToPostOrderCGSCCPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x222caf1)
#19 0x000056453146d4a4 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x21304a4)
#20 0x000056453156c6ff llvm::ModuleInlinerWrapperPass::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x222f6ff)
#21 0x0000564532a8f1c1 llvm::detail::PassModel<llvm::Module, llvm::ModuleInlinerWrapperPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x37521c1)
#22 0x000056453146d4a4 llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/bed22/cheri/output/sdk/bin/clang+++0x21304a4)
#23 0x0000564531d955f8 (anonymous namespace)::EmitAssemblyHelper::EmitAssemblyWithNewPassManager(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) BackendUtil.cpp:0:0
#24 0x0000564531d9a4dc clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/home/bed22/cheri/output/sdk/bin/clang+++0x2a5d4dc)
#25 0x0000564532a24b4e clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/bed22/cheri/output/sdk/bin/clang+++0x36e7b4e)
#26 0x0000564532ddac39 clang::ParseAST(clang::Sema&, bool, bool) (/home/bed22/cheri/output/sdk/bin/clang+++0x3a9dc39)
#27 0x0000564532a24ce8 clang::CodeGenAction::ExecuteAction() (/home/bed22/cheri/output/sdk/bin/clang+++0x36e7ce8)
#28 0x00005645323b5c01 clang::FrontendAction::Execute() (/home/bed22/cheri/output/sdk/bin/clang+++0x3078c01)
#29 0x00005645323503f2 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/bed22/cheri/output/sdk/bin/clang+++0x30133f2)
#30 0x0000564532481978 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/bed22/cheri/output/sdk/bin/clang+++0x3144978)
#31 0x0000564530221c4e cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/bed22/cheri/output/sdk/bin/clang+++0xee4c4e)
#32 0x000056453021c4c9 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#33 0x0000564532207955 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const::'lambda'()>(long) Job.cpp:0:0
#34 0x0000564531a4c983 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/home/bed22/cheri/output/sdk/bin/clang+++0x270f983)
#35 0x00005645322082f4 clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const (.part.162) Job.cpp:0:0
#36 0x00005645321dfeba clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/home/bed22/cheri/output/sdk/bin/clang+++0x2ea2eba)
#37 0x00005645321e0b5f clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/home/bed22/cheri/output/sdk/bin/clang+++0x2ea3b5f)
#38 0x00005645321e956a clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/home/bed22/cheri/output/sdk/bin/clang+++0x2eac56a)
#39 0x0000564530178c80 main (/home/bed22/cheri/output/sdk/bin/clang+++0xe3bc80)
#40 0x00007fd5ad2d0bf7 __libc_start_main /build/glibc-S9d2JN/glibc-2.27/csu/../csu/libc-start.c:344:0
#41 0x000056453021c00a _start (/home/bed22/cheri/output/sdk/bin/clang+++0xedf00a)
clang-13: error: clang frontend command failed with exit code 134 (use -v to see invocation)
clang version 13.0.0 (https://github.com/CTSRD-CHERI/llvm-project.git ea3e23aa96b1e39e8473d4e93b773243ba910e58)
Target: riscv64-unknown-freebsd13.0
Thread model: posix
InstalledDir: /home/bed22/cheri/output/sdk/bin
clang-13: note: diagnostic msg: 
********************
PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-13: note: diagnostic msg: /tmp/bed22/libunwind-209b2f.cpp
clang-13: note: diagnostic msg: /tmp/bed22/libunwind-209b2f.sh
clang-13: note: diagnostic msg: 
********************
--- libunwind.o ---
*** [libunwind.o] Error code 134
```

**Comment by kwitaszczyk** [comment]
I also got this crash: [libunwind-6a2b7e.{cpp,sh}](https://gist.github.com/kwitaszczyk/b6e1ebbe4444a1942c3ef88c5474ee8e).
```
===> lib/libgcc_eh (obj,all,install) 
Assertion failed: (BitWidth == DL.getIndexTypeSizeInBits(getType()) && "The offset bit width does not match the DL specification."), function stripAndAccumulateConstantOffsets, file /home/kw543/personal/envs/latest/llvm-project/llvm/lib/IR/Value.cpp, line 696.
 #0 0x00000000033cc288 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x33cc288)
 #1 0x00000000033cc688 __assert (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x33cc688)
 #2 0x0000000002d2feca llvm::Value::stripAndAccumulateConstantOffsets(llvm::DataLayout const&, llvm::APInt&, bool, llvm::function_ref<bool (llvm::Value&, llvm::APInt&)>) const (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d2feca)
 #3 0x00000000033f48d5 llvm::tryPromoteCall(llvm::CallBase&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x33f48d5)
 #4 0x0000000002e6d412 llvm::InlinerPass::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2e6d412)
 #5 0x0000000002e71422 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::InlinerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) Inliner.cpp:0:0
 #6 0x0000000002d90e44 llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d90e44)
 #7 0x0000000002e71692 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::PassManager<llvm::LazyCallGraph::SCC, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) Inliner.cpp:0:0
 #8 0x0000000002d93faf llvm::DevirtSCCRepeatedPass::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d93faf)
 #9 0x0000000002e71b72 llvm::detail::PassModel<llvm::LazyCallGraph::SCC, llvm::DevirtSCCRepeatedPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&>::run(llvm::LazyCallGraph::SCC&, llvm::AnalysisManager<llvm::LazyCallGraph::SCC, llvm::LazyCallGraph&>&, llvm::LazyCallGraph&, llvm::CGSCCUpdateResult&) Inliner.cpp:0:0
#10 0x0000000002d92be3 llvm::ModuleToPostOrderCGSCCPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d92be3)
#11 0x0000000002e719b2 llvm::detail::PassModel<llvm::Module, llvm::ModuleToPostOrderCGSCCPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) Inliner.cpp:0:0
#12 0x0000000002d1115a llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d1115a)
#13 0x0000000002e6f30d llvm::ModuleInlinerWrapperPass::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2e6f30d)
#14 0x0000000004553502 llvm::detail::PassModel<llvm::Module, llvm::ModuleInlinerWrapperPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) PassBuilder.cpp:0:0
#15 0x0000000002d1115a llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module> >::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x2d1115a)
#16 0x000000000361d5c5 (anonymous namespace)::EmitAssemblyHelper::EmitAssemblyWithNewPassManager(clang::BackendAction, std::__1::unique_ptr<llvm::raw_pwrite_stream, std::__1::default_delete<llvm::raw_pwrite_stream> >) BackendUtil.cpp:0:0
#17 0x00000000036187ec clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::DataLayout const&, llvm::Module*, clang::BackendAction, std::__1::unique_ptr<llvm::raw_pwrite_stream, std::__1::default_delete<llvm::raw_pwrite_stream> >) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x36187ec)
#18 0x0000000003e9db6f clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) CodeGenAction.cpp:0:0
#19 0x00000000046c7113 clang::ParseAST(clang::Sema&, bool, bool) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x46c7113)
#20 0x0000000003ded6e3 clang::FrontendAction::Execute() (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3ded6e3)
#21 0x0000000003d29194 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3d29194)
#22 0x0000000003e9710c clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3e9710c)
#23 0x0000000001bb9be9 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x1bb9be9)
#24 0x0000000001bb7549 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#25 0x0000000003bf0d47 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*, bool*) const::$_1>(long) Job.cpp:0:0
#26 0x0000000003343090 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3343090)
#27 0x0000000003bf085b clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >*, bool*) const (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3bf085b)
#28 0x0000000003bbd1ed clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3bbd1ed)
#29 0x0000000003bbd7b7 clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) const (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3bbd7b7)
#30 0x0000000003bd332c clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::__1::pair<int, clang::driver::Command const*> >&) (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x3bd332c)
#31 0x0000000001bb6cfd main (/home/kw543/personal/envs/latest/output/sdk/bin/clang+++0x1bb6cfd)
```

**Comment by jrtc27** [comment]
This is a build configuration issue. Historically NEWPM was opt-in at run-time, then a CMake option was added to configure the default but that still itself defaulted to OFF, then that was changed to ON. If you build all three states in sequence then you end up with a tree that disables NEWPM by default, as is required for CHERI-LLVM (or, was; after the latest upstream merge in dev Alex fixed this bug), but if you have an old LLVM tree that you then do a non-clean build on that skips the intermediate state, you end up with NEWPM being on by default, and hit this bug (which can be reproduced on all toolchains with `-fexperimental-new-pass-manger` or `-fno-legacy-pass-manager`, depending on the version).

TL;DR: Anyone hitting this should do a clean rebuild.
